homeassistant/components/qbus/climate.py

```diff
@@ -0,0 +1,159 @@
+    added_outputs: list[QbusMqttOutput] = []
+
+    def _check_outputs() -> None:
+        add_new_outputs(
+            coordinator,
+            added_outputs,
+            lambda output: output.type == "thermo",
+            QbusClimate,
+            async_add_entities,
+        )
+
+    _check_outputs()
+    entry.async_on_unload(coordinator.async_add_listener(_check_outputs))
+
+
+class QbusClimate(QbusEntity, ClimateEntity):
+    """Representation of a Qbus climate entity."""
+
+    _attr_hvac_modes = [HVACMode.OFF, HVACMode.HEAT]
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+
+    def __init__(self, mqtt_output: QbusMqttOutput) -> None:
+        """Initialize climate entity."""
+
+        super().__init__(mqtt_output)
+
+        self._attr_hvac_action = HVACAction.IDLE
+        self._attr_hvac_mode = HVACMode.OFF
+
+        set_temp: dict = mqtt_output.properties.get(KEY_PROPERTIES_SET_TEMPERATURE, {})
+        current_regime: dict = mqtt_output.properties.get(KEY_PROPERTIES_REGIME, {})
+
+        self._attr_min_temp = set_temp.get("min", 0)
+        self._attr_max_temp = set_temp.get("max", 35)
+        self._attr_target_temperature_step = set_temp.get("step", 0.5)
+        self._attr_preset_modes = current_regime.get("enumValues", [])
+        self._attr_preset_mode = (
+            self._attr_preset_modes[0] if len(self._attr_preset_modes) > 0 else ""
+        )
+
+        self._cancel_state_request: CALLBACK_TYPE | None = None
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new target preset mode."""
+        state = QbusMqttThermoState(id=self._mqtt_output.id, type=StateType.STATE)
+        state.write_regime(preset_mode)
+
+        await self._async_publish_output_state(state)
+        self._attr_preset_mode = preset_mode
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+
+        if temperature is not None and isinstance(temperature, float):
+            state = QbusMqttThermoState(id=self._mqtt_output.id, type=StateType.STATE)
+            state.write_set_temperature(temperature)
+
+            await self._async_publish_output_state(state)
+            self._attr_target_temperature = temperature
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        # It is not supported to explicitly set the HVAC mode. The value
+        # is determined automatically.
+
+    async def _state_received(self, msg: ReceiveMessage) -> None:
+        state = self._message_factory.parse_output_state(
+            QbusMqttThermoState, msg.payload
+        )
+
+        if state is None:
+            return
+
+        self._attr_preset_mode = state.read_regime() or self._attr_preset_mode
```

Comment:
```suggestion
        if (preset_mode := state.read_regime()):
            self._attr_preset_mode = preset_mode
```
Makes it a bit more explicit.

Reply:
Same for the other 2 attr bellow